discretize/DiffOperators.py

```diff
@@ -302,21 +447,57 @@ def setCellGradBC(self, BC):
         self._cellGradBC = None
         self._cellGradBC_list = BC
         return BC
-    _cellGradBC_list = 'neumann'
 
+    @property
+    def _cellGradxStencil(self):
+        # TODO: remove this hard-coding
+        BC = ['neumann', 'neumann']
+        if self.dim == 1:
+            G1 = ddxCellGrad(self.nCx, BC)
+        elif self.dim == 2:
+            G1 = sp.kron(speye(self.nCy), ddxCellGrad(self.nCx, BC))
+        elif self.dim == 3:
+            G1 = kron3(
+                speye(self.nCz),
+                speye(self.nCy),
+                ddxCellGrad(self.nCx, BC)
+            )
+        return G1
+
+    @property
+    def _cellGradyStencil(self):
+        if self.dim < 2:
+            return None
+        BC = ['neumann', 'neumann'] # TODO: remove this hard-coding
+        n = self.vnC
+        if(self.dim == 2):
+            G2 = sp.kron(ddxCellGrad(n[1], BC), speye(n[0]))
+        elif self.dim == 3:
+            G2 = kron3(speye(n[2]), ddxCellGrad(n[1], BC), speye(n[0]))
+        return G2
+
+    @property
+    def _cellGradzStencil(self):
+        if self.dim < 3:
+            return None
+        BC = ['neumann', 'neumann'] # TODO: remove this hard-coding
+        n = self.vnC
+        G3 = kron3(ddxCellGrad(n[2], BC), speye(n[1]), speye(n[0]))
+        return G3
+
+    @property
```

Comment:
I made this a property. Previously it was a function with no inputs, but it makes more sense as a property. This will require an update in the regularization mesh over in SimPEG